cpp/include/cudf/copying.hpp

```diff
@@ -44,6 +44,21 @@ gdf_column empty_like(gdf_column const& input);
  */
 gdf_column allocate_like(gdf_column const& input, bool allocate_mask_if_exists = true, cudaStream_t stream = 0);
 
+/**
+ * @brief Allocates a new column of the specified size and type.
+ * 
+ * @param dtype Datatype of the column
+ * @param size Size of the column
+ * @param allocate_mask Optional Whether or not to allocate bitmask
+ * @param info Optional Extra info for the column
+ * @param stream Optional stream in which to perform allocation
+ * @return gdf_column An allocated column of given size and type
+ */
+gdf_column allocate_column(gdf_dtype dtype, gdf_size_type size,
```

Comment:
That overload of `allocate_like` still takes the datatype from the input column. I was hoping to also use this `allocate_column` in #2156 [here](https://github.com/rapidsai/cudf/pull/2156/files#diff-98c23212fa6fcce605601c29795bb281R75). Over there I do need to allocate a column of a dtype that is not guaranteed to appear in input.

Reply:
OK, but I think this belongs in include/legacy/column.hpp (since it's not a copy and it will go away once cudf::column is deployed), not in copying.hpp.